sdk/monitor/microsoft-opentelemetry-exporter-azuremonitor/microsoft/opentelemetry/exporter/azuremonitor/_storage.py

```diff
@@ -166,6 +155,12 @@ def get(self):
         return None
 
     def put(self, data, lease_period=0):
+        # Create path if it doesn't exist
+        try:
+            if not os.path.isdir(self.path):
```

Comment:
How come we are moving this into here?

Reply:
Path creation was done during initialization and every minute after that before, we don't provide the path configuration during initialization anymore so to be able to change the folder path we need to check all the time now, not sure why this logic was in minute intervals before so maybe I'm missing something.

Reply:
I believe it was put in `_maintenance_routine` to guarantee that it would always exist whenever `gets()` is called right after. Perhaps we have to put the check everywhere now that uses path.